packages/app/lib/ui/widgets/note_field.dart

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import 'package:app/ui/widgets/card.dart';
+import 'package:app/ui/widgets/card_action_button.dart';
+import 'package:app/ui/widgets/card_header.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+typedef OnUpdate = void Function(String?);
+
+enum InputMode { read, edit }
+
+class NoteField extends StatefulWidget {
+  final String? value;
+  final OnUpdate onUpdate;
+
+  NoteField(this.value, {super.key, required this.onUpdate});
+
+  // @override
+  State<NoteField> createState() => _NoteFieldState();
+}
+
+class _NoteFieldState extends State<NoteField> {
+  InputMode _inputMode = InputMode.read;
+  late TextEditingController _textInputController;
+  late FocusNode _textInputFocusNode;
+  final _formKey = GlobalKey<FormState>();
+
+  @override
+  void initState() {
+    super.initState();
+    _textInputController = TextEditingController(text: widget.value ?? "");
+    _textInputFocusNode =
+        FocusNode(canRequestFocus: _inputMode == InputMode.edit);
+  }
+
+  @override
+  void didUpdateWidget(prev) {
+    super.didUpdateWidget(prev);
+    // Needed in order "reset" the initialValue of the input based on the change received by widget.value
+    _textInputController.text = widget.value ?? "";
+  }
+
+  @override
+  void dispose() {
+    _textInputController.dispose();
+    _textInputFocusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final t = AppLocalizations.of(context)!;
+
+    final initialValue = widget.value ?? '';
+
+    return Form(
+        key: _formKey,
+        child: MeliCard(
+            child: Column(children: [
+          MeliCardHeader(
+              title: t.noteCardTitle,
+              icon: _inputMode == InputMode.read
+                  ? CardActionButton(
+                      icon: Icon(Icons.edit),
+                      onPressed: _handleStartEditMode,
+                    )
+                  : null),
+          Container(
+              padding: EdgeInsets.symmetric(vertical: 10.0, horizontal: 6.0),
+              alignment: AlignmentDirectional.centerStart,
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  // TODO: TextField seems to have a bug where it does not respect _textInputFocusNode.canRequestFocus being false when in 'read' input mode
```

Comment:
just to add some more context, the bug that manifests from this is the input field showing the focused state after tapping it _while in read mode_. 

`TextField.caRequestFocus` actually worked for this specific bug, but:

1. It's [deprecated](https://docs.flutter.dev/release/breaking-changes/can-request-focus) (although not in stable yet) in favor of the focus node approach (which I'm claiming is broken)

2. Resulted in a worse bug where there seemed to be a race condition that prevented the desired focusable state when toggling between read and edit modes.

Reply:
Maybe `TextFormField` has some children which still allow being focused, as apparently `canRequestFocus` does not necessarily control that:

> If set to false on a [FocusNode](https://api.flutter.dev/flutter/widgets/FocusNode-class.html), it will not affect the focusability of children of the node.
(from https://api.flutter.dev/flutter/widgets/FocusNode/canRequestFocus.html)